### PR TITLE
Make validation tests specify validation rules explicitly to avoid running all

### PR DESCRIPTION
**Pull request by jzhu13**

### CODE REVIEW

File: metricflow/test/model/validations/test_validity_param_definitions.py [hunk 1]
```diff
@@ -70,6 +70,7 @@ def test_validity_window_must_have_a_start() -> None:
     validity_window_file = YamlConfigFile(filepath="inline_for_test", contents=yaml_contents)
     model = parse_yaml_files_to_validation_ready_model([base_model_file(), validity_window_file])
 
+    #foo bar test commit
```
Comment (tlento) [inline metricflow/test/model/validations/test_validity_param_definitions.py:73]: 🤪 

(For real though, looking forward to the update or new PR with the proper changes.)

**Comment by jzhu13** [comment]
Test duration after the changes in this PR (from running 'poetry run pytest  --durations=0 metricflow/test/model/validations/'): 

4.99s call     metricflow/test/model/validations/test_element_const.py::test_cross_element_names
3.71s call     metricflow/test/model/validations/test_configurable_rules.py::test_can_configure_model_validator_rules
3.69s call     metricflow/test/model/validations/test_measures.py::test_measures_only_exist_in_one_data_source
3.08s call     metricflow/test/model/validations/test_unique_valid_name.py::test_top_level_metric_can_have_same_name_as_any_other_top_level_item
2.18s call     metricflow/test/model/validations/test_identifiers.py::test_data_source_cant_have_more_than_one_primary_identifier
2.14s call     metricflow/test/model/validations/test_validity_param_definitions.py::test_validity_window_uses_two_dimensions
2.12s call     metricflow/test/model/validations/test_validity_param_definitions.py::test_two_dimension_validity_windows_must_not_overload_start_and_end
1.94s call     metricflow/test/model/validations/test_agg_time_dimension.py::test_missing_primary_time_ok_if_all_measures_have_agg_time_dim
1.90s call     metricflow/test/model/validations/test_measures.py::test_metric_missing_measure
1.89s call     metricflow/test/model/validations/test_identifiers.py::test_mismatched_identifier
1.87s call     metricflow/test/model/validations/test_validity_param_definitions.py::test_validity_window_must_have_a_natural_key
1.87s call     metricflow/test/model/validations/test_validity_param_definitions.py::test_validity_window_does_not_use_primary_key
1.87s call     metricflow/test/model/validations/test_validity_param_definitions.py::test_multiple_validity_windows_are_invalid
1.83s call     metricflow/test/model/validations/test_identifiers.py::test_composite_identifiers_ref_and_name
1.82s call     metricflow/test/model/validations/test_measures.py::test_measure_alias_measure_name_conflict
1.82s call     metricflow/test/model/validations/test_validity_param_definitions.py::test_empty_validity_windows_are_invalid
1.82s call     metricflow/test/model/validations/test_identifiers.py::test_composite_identifiers_nonexistent_ref
1.81s call     metricflow/test/model/validations/test_validity_param_definitions.py::test_measures_are_prevented
1.80s call     metricflow/test/model/validations/test_dimension_const.py::test_incompatible_dimension_type
1.71s call     metricflow/test/model/validations/test_measures.py::test_invalid_non_additive_dimension_properties
1.68s call     metricflow/test/model/validations/test_unique_valid_name.py::test_duplicate_data_source_name
1.68s call     metricflow/test/model/validations/test_dimension_const.py::test_incompatible_dimension_is_partition
1.68s call     metricflow/test/model/validations/test_identifiers.py::test_invalid_composite_identifiers
1.65s call     metricflow/test/model/validations/test_unique_valid_name.py::test_duplicate_metric_name
1.64s call     metricflow/test/model/validations/test_unique_valid_name.py::test_duplicate_materalization_name
1.63s call     metricflow/test/model/validations/test_dimension_const.py::test_multiple_primary_time_dimensions
1.63s call     metricflow/test/model/validations/test_unique_valid_name.py::test_duplicate_identifier_name
1.62s call     metricflow/test/model/validations/test_unique_valid_name.py::test_duplicate_dimension_name
1.61s call     metricflow/test/model/validations/test_measures.py::test_measure_alias_is_set_when_required
1.61s call     metricflow/test/model/validations/test_metrics.py::test_metric_multiple_primary_time_dims
1.59s call     metricflow/test/model/validations/test_unique_valid_name.py::test_top_level_elements_cant_share_names_except_with_metrics
1.58s call     metricflow/test/model/validations/test_validity_param_definitions.py::test_validity_window_configuration
1.58s call     metricflow/test/model/validations/test_identifiers.py::test_natural_identifier_used_in_wrong_context
1.58s call     metricflow/test/model/validations/test_agg_time_dimension.py::test_unset_aggregation_time_dimension
1.57s call     metricflow/test/model/validations/test_measures.py::test_invalid_measure_alias_name
1.55s call     metricflow/test/model/validations/test_measures.py::test_reused_measure_alias_within_metric
1.55s call     metricflow/test/model/validations/test_metrics.py::test_generated_metrics_only
1.54s call     metricflow/test/model/validations/test_measures.py::test_reused_measure_alias
1.54s call     metricflow/test/model/validations/test_identifiers.py::test_multiple_natural_identifiers
1.53s call     metricflow/test/model/validations/test_metrics.py::test_derived_metric
1.52s call     metricflow/test/model/validations/test_unique_valid_name.py::test_duplicate_measure_name
1.50s call     metricflow/test/model/validations/test_measures.py::test_count_measure_missing_expr
1.50s call     metricflow/test/model/validations/test_metrics.py::test_metric_no_time_dim
1.44s call     metricflow/test/model/validations/test_measures.py::test_count_measure_with_distinct_expr
1.44s call     metricflow/test/model/validations/test_metrics.py::test_metric_no_time_dim_dim_only_source
1.44s call     metricflow/test/model/validations/test_agg_time_dimension.py::test_invalid_aggregation_time_dimension
1.39s call     metricflow/test/model/validations/test_validity_param_definitions.py::test_validity_window_must_have_a_start
1.39s call     metricflow/test/model/validations/test_validity_param_definitions.py::test_validity_window_must_have_an_end
0.52s call     metricflow/test/model/validations/test_data_warehouse_tasks.py::test_validate_metrics
0.31s call     metricflow/test/model/validations/test_materializations.py::test_missing_primary_time_dimension
0.28s setup    metricflow/test/model/validations/test_configurable_rules.py::test_can_configure_model_validator_rules
0.27s call     metricflow/test/model/validations/test_materializations.py::test_invalid_time_granularity
0.24s call     metricflow/test/model/validations/test_materializations.py::test_invalid_metric_name
0.24s setup    metricflow/test/model/validations/test_agg_time_dimension.py::test_invalid_aggregation_time_dimension
0.22s call     metricflow/test/model/validations/test_materializations.py::test_identifier
0.21s call     metricflow/test/model/validations/test_materializations.py::test_valid_time_granularity
0.20s call     metricflow/test/model/validations/test_materializations.py::test_invalid_dimension_name
0.20s call     metricflow/test/model/validations/test_data_warehouse_tasks.py::test_build_metric_tasks
0.17s setup    metricflow/test/model/validations/test_data_warehouse_tasks.py::test_build_data_source_tasks
0.16s call     metricflow/test/model/validations/test_materializations.py::test_materialization_validation
0.05s setup    metricflow/test/model/validations/test_data_warehouse_tasks.py::test_validate_data_sources
0.03s setup    metricflow/test/model/validations/test_agg_time_dimension.py::test_unset_aggregation_time_dimension
0.03s call     metricflow/test/model/validations/test_data_warehouse_tasks.py::test_validate_dimensions
0.02s call     metricflow/test/model/validations/test_data_warehouse_tasks.py::test_validate_measures
0.02s call     metricflow/test/model/validations/test_data_warehouse_tasks.py::test_task_runner
0.02s call     metricflow/test/model/validations/test_data_warehouse_tasks.py::test_validate_identifiers
0.02s call     metricflow/test/model/validations/test_reserved_keywords.py::test_reserved_keywords_in_composite_identifiers
0.02s call     metricflow/test/model/validations/test_reserved_keywords.py::test_reserved_keywords_in_measures
0.01s call     metricflow/test/model/validations/test_data_warehouse_tasks.py::test_validate_data_sources
0.01s call     metricflow/test/model/validations/test_reserved_keywords.py::test_reserved_keywords_in_sql_table
0.01s call     metricflow/test/model/validations/test_reserved_keywords.py::test_reserved_keywords_in_identifiers
0.01s call     metricflow/test/model/validations/test_reserved_keywords.py::test_reserved_keywords_in_dimensions
0.01s teardown metricflow/test/model/validations/test_validity_param_definitions.py::test_validity_window_does_not_use_primary_key

**Review by QMalcolm — COMMENTED** [review]
The changes look good! I parsed the above posted stats and found the following:

> Old tests: 75 tests took a total of 105.77 seconds
New tests: 73 tests took a total of 92.80 seconds

Progress! We made gains, but not by as much as we hoped (I think). As a side note though, there were 2 fewer test actions in the second set of stats, which seems odd? I did not notice any deleted tests 🤔

I wonder if we can take this further though? There must be a bottle neck somewhere. My first thought goes to [ModelValidator](https://github.com/transform-data/metricflow/blob/f309c74617a0688b8921f6d24ad002888e2d3b3d/metricflow/model/model_validator.py#L40). It uses a [ProcessPoolExecutor](https://github.com/transform-data/metricflow/blob/f309c74617a0688b8921f6d24ad002888e2d3b3d/metricflow/model/model_validator.py#L80) to run the model through the rules. This is useful because some people put absolutely massive models through MetricFlow and the number of validation rules MetricFlow has generally only grows, but it's non negligible overhead. In this instance we're running a relatively small model through only one rule in any given test. Given that we're really testing the individual rules at this point, and not the `ModelValidator` class, perhaps we could make some gains by running the individual rules directly instead of via the `ModelValidator`. This would also mean that we'd have to check the returned validation issues instead of matching against an exception though (which has pros and cons).

**Review by QMalcolm — APPROVED** [review]
I have no additional ideas to speed this up further. Let's run with this improvement. Thanks for doing this work!

**Pull request merged**